### DISCUSSION TITLE
Cannot connect to Redis when i use mokey_patch

### DISCUSSION THREAD

**Discussion by ZanhNe** [miguelgrinberg/Flask-SocketIO#2121] [Q&A] [answered]
`import eventlet`
`eventlet.monkey_patch()` 

`import redis`

`redis_client = redis.StrictRedis(host='localhost', port=6379, charset='utf-8', decode_responses=True)`

`try:
    redis_client.ping()
    print("Redis connected")
except Exception as e:
    print("Cannot connect to Redis:", e)
   `
  
Someone please help me, I was planning to use monkey_patch() to be able to use SocketIO's message_queue, but adding monkey_patch() I can't connect to Redis server "Cannot connect to Redis: Error 11001 connecting to localhost:6379. No address found."
   
![image](https://github.com/user-attachments/assets/91e98b49-c122-484d-a399-0fb719b15965)
But if I remove monkey_patch(), the connection is still possible.
![image](https://github.com/user-attachments/assets/e2c68976-a48b-4fe7-a02d-cba53a87307c)

**Comment by miguelgrinberg** [comment, marked as answer]
I'm not really sure how I can help with this, as it is unrelated to Flask-SocketIO. If you are sure that you have Redis running and you have confirmed that you can connect to it without eventlet, then this could be an eventlet bug. You can try asking in the Eventlet project, but keep in mind Eventlet is not in active development anymore, so it isn't a great idea to use it on new projects.

**Reply by ZanhNe** [reply]
Thank you very much, it's true that when I switched to gevent, the error was gone. I also don't understand why eventlet is like that and there is still no proposal to remove it.
